src/classes/BDI_MappingServiceAdvanced.cls

```diff
@@ -415,4 +437,31 @@ public with sharing class BDI_MappingServiceAdvanced implements BDI_MappingServi
 
         return fieldMapping;
     }
+
+    private void constructBrokenFieldMappings (Data_Import_Field_Mapping__mdt difm) {
```

Comment:
Any particular reason we didn't want to perform this type of logic inside the `BDI_FieldMapping` constructor?